library_api/dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM golang:1.19-buster AS build
+
+WORKDIR /library_api
+
+# pre-copy/cache go.mod for pre-downloading dependencies and only redownloading them in subsequent builds if they change
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . ./
+
+RUN go build -o main .
+
+EXPOSE 8000
+
+CMD ["./main"]
```

Comment:
use `docker inspect [image name]` to check how much your image weight, then use multistage build and check it again -> you will notice huge memory improvement

Reply:
after multi-stage build 
![image](https://user-images.githubusercontent.com/37672653/234219122-16953d84-5772-4a1c-852f-fdb1b4ef1e67.png)